portals/devportal/source/src/app/components/Apis/Details/GraphQLConsole/AdditionalHeaders.jsx

```diff
@@ -0,0 +1,265 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { makeStyles } from '@material-ui/core/styles';
+import { FormattedMessage } from 'react-intl';
+import ExpansionPanel from '@material-ui/core/ExpansionPanel';
+import ExpansionPanelDetails from '@material-ui/core/ExpansionPanelDetails';
+import ExpansionPanelSummary from '@material-ui/core/ExpansionPanelSummary';
+import Typography from '@material-ui/core/Typography';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import Box from '@material-ui/core/Box';
+import Table from '@material-ui/core/Table';
+import TableBody from '@material-ui/core/TableBody';
+import TableCell from '@material-ui/core/TableCell';
+import TableHead from '@material-ui/core/TableHead';
+import TableRow from '@material-ui/core/TableRow';
+import cloneDeep from 'lodash.clonedeep';
+import AddEditAdditionalHeaders from 'AppComponents/Apis/Details/GraphQLConsole/AddEditAdditionalHeaders';
+import DeleteHeader from 'AppComponents/Apis/Details/GraphQLConsole/DeleteHeader';
+
+const useStyles = makeStyles((theme) => ({
+    root: {
+        width: '100%',
+        marginBottom: 20,
+    },
+    heading: {
+        fontSize: theme.typography.pxToRem(15),
+        flexBasis: '33.33%',
+        flexShrink: 0,
+        flex: 1,
+        alignItems: 'center',
+    },
+    secondaryHeading: {
+        fontSize: theme.typography.pxToRem(15),
+        color: theme.palette.text.secondary,
+        display: 'flex',
+        alignItems: 'center',
+    },
+    table: {
+        marginBottom: 40,
+        background: '#efefef',
+        '& th': {
+            background: '#ccc',
+        },
+    },
+    expandContentRoot: {
+        flexDirection: 'column',
+    },
+    subsubTitle: {
+        fontSize: '0.81rem',
+    },
+    alert: {
+        flex: 1,
+    },
+    hr: {
+        border: 'solid 1px #efefef',
+        width: '100%',
+    },
+    descriptionBox: {
+        marginLeft: theme.spacing(1),
+    },
+}));
+
+/**
+ * Create UUID
+ * @returns {string} random uuid string.
+ */
+function getUUID() {
+    return Math.random().toString(36).substring(2) + Date.now().toString(36);
+}
+/**
+ * Render a list
+ * @returns {JSX} Header AppBar components.
+ * @param {JSON} props Provides props from parent
+ */
+function AdditionalHeaders(props) {
+    const classes = useStyles();
+    const { additionalHeaders, setAdditionalHeaders } = props;
+    const [expanded, setExpanded] = React.useState(false);
+
+    const handleChange = (panel) => (event, isExpanded) => {
+        setExpanded(isExpanded ? panel : false);
+    };
+    const getNewItem = () => {
+        return ({
+            id: '',
+            name: '',
+            value: '',
+        });
+    };
+
+    const addItem = (item) => {
+        const { name, value } = item;
+        const newItem = getNewItem();
+        newItem.id = getUUID();
```

Comment:
Why we need a UUID ?

Reply:
We need some id to track each item to correctly  update/delete item. L122, L133

Reply:
I think this is bit overcomplicated ,
Can you pls have a look , we can use an object track additional header key value pair
https://codesandbox.io/s/addedit-json-object-vnx7c?file=/demo.js